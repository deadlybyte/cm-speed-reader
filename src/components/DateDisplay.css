.CM-date {
  color: #84ffff;
  display: inline-block;
  text-align: left;
}
