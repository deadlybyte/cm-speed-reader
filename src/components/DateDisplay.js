// @flow
import React from 'react';
import moment from 'moment';
import './DateDisplay.css';

const DateDisplay = () => {
  const today = moment().format('dddd Do MMMM YYYY');
  return (
    <div className="CM-overlay CM-date">
      <span>{today}</span>
    </div>
  );
};

export default DateDisplay;
