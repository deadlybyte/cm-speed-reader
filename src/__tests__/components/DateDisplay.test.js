import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import DateDisplay from '../../components/DateDisplay';

describe('DateDisplay', () => {
  const now = Date.now;

  beforeEach(() => {
    Date.now = jest.fn(() => 1487076708000);
  });

  afterEach(() => {
    Date.now = now;
  });

  describe('render', () => {
    it('should render without crashing', () => {
      const div = document.createElement('div');
      ReactDOM.render(<DateDisplay />, div);
      ReactDOM.unmountComponentAtNode(div);
    });

    it('should render as expected', () => {
      const componentUnderTest = renderer.create(<DateDisplay />)
        .toJSON();
      expect(componentUnderTest).toMatchSnapshot();
    });
  });
});
