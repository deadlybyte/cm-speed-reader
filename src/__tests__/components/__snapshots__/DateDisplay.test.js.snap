// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateDisplay render should render as expected 1`] = `
<div
  className="CM-overlay CM-date"
>
  <span>
    Tuesday 14th February 2017
  </span>
</div>
`;
